### DISCUSSION TITLE
Why copyObject of S3AsyncClient result file with 0 Byte

### DISCUSSION THREAD

**Discussion by SupawiriyaSaranarak** [aws/aws-sdk-java-v2#3759] [Q&A]
I try to copy file from one directory to another in the same bucket in java. It seems work but when I see the file size it shows 0 byte. 
I use spring webfulx and S3AsyncClient to work on it

try {
            encodedUrl = URLEncoder.encode(bucket + "/" + previousObjectKey, StandardCharsets.UTF_8.toString());

        } catch (UnsupportedEncodingException e) {
            log.info("URL could not be encoded: {}", e.getMessage());
            return Mono.error(new StorageHelperException(e.getMessage(), e));
        }

        var copyRequest = CopyObjectRequest.builder()
                         .copySourceIfMatch(encodedUrl)
                        .destinationBucket(bucket)
                        .destinationKey(newObjectKey).build();

s3AsyncClient.copyObject(copyRequest);

and I have tried both encoded and non-encoded already

**Comment by debora-ito** [comment]
The original object size was greater than 0 bytes, I assume.

Can you share an example of the object tree, so I can try to reproduce?

Also, if you can reproduce it in a local environment, can you enable the verbose wirelogs? Please make sure to redact any sensitive information the logs may contain (like access keys).
For instructions on how to enable wirelogs see: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/logging-slf4j.html#sdk-java-logging-verbose
